### PR TITLE
Groups insertifnotexists

### PR DESCRIPTION
**Pull request by DeepDiver1975**
alternative for https://github.com/owncloud/core/pull/21708

### CODE REVIEW

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works - code looks good :+1:

File: lib/private/group/database.php [hunk 1]
```diff
@@ -85,44 +85,15 @@ private function fixDI() {
 	public function createGroup( $gid ) {
 		$this->fixDI();
 
-		// Check cache first
-		if (isset($this->groupCache[$gid])) {
```
Comment (rullzer) [inline lib/private/group/database.php]: You deleted a bit to much. First check the cache...

Comment (nickvergessen) [inline lib/private/group/database.php]: I intentionally deleted it, we don't create groups that often.
So I thought better try to add it and see if that works.

The check may give a wrong result, when the group was deleted in a parallel request. It thinks the group already exists doesnt add it, although it doesnt exist anymore...

Comment (DeepDiver1975) [inline lib/private/group/database.php]: quite valid ...

Comment (rullzer) [inline lib/private/group/database.php]: ah mm. fine.. indeed does not happen that often.

But paralllel request chaning such things might yield unexpected results... (with or without caching).

**Pull request merged**